crates/revm/src/handler/mainnet/post_execution.rs

```diff
@@ -36,11 +37,27 @@ pub fn reward_beneficiary<EvmWiringT: EvmWiring, SPEC: Spec>(
     // transfer fee to coinbase/beneficiary.
     // EIP-1559 discard basefee for coinbase transfer. Basefee amount of gas is discarded.
     let coinbase_gas_price = if SPEC::enabled(LONDON) {
-        effective_gas_price.saturating_sub(*context.evm.env.block.basefee())
+        let basefee = *context.evm.env.block.basefee();
+
+        // Skip the transfer if basefee and max priority fee per gas is zero, like geth.
+        if basefee.is_zero()
```

Comment:
Why do we skip if basefee is zero? Shouldn't we skip only if `coinbase_gas_price == 0`

Reply:
It looks like that's `geth`'s heuristic: https://github.com/ethereum/go-ethereum/blob/ae707445f54972f4fddc7c6cf8ba2de083636e50/core/state_transition.go#L468